### PR TITLE
Solve DNS not resolve after network comes up.

### PR DESCRIPTION
**Pull request by papampi**
While name resolution not working check ping and restart 3main if ping is ok.

### CODE REVIEW

**Comment by LuKePicci** [comment]
Not sure how this can solve the problem. I mean, how restarting an application could restore system-wide name resolution functionality? Every time the` nc google.com 443` loop detect a connection problem it restart a new instance of the netcat program, so it seems restarting netcat don't let netcat get network resolution fixed. How restarting the miner application could solve name resolution issues affecting netcat? I would like to see the bug bulletin @doftorul was referring to in #94 .

**Comment by papampi** [comment]
I tested it on 5 rigs, no idea whats the problem, when rigs stuck on dns resolve issue I killed miner manually after 2-3 minutes dns gets resolved.

**Comment by papampi** [comment]
Will send you some logs via email

**Comment by LuKePicci** [comment]
I would like to know if a local dnsmasq caching instance is able to prevent this kind of issues or it stucks the same way.

edit: I see dnsmasq is there already, have you attempted to restart that service instead of restarting miner?

**Comment by papampi** [comment]
Yup, restarted network-manager as well, no help if miner still running.

**Comment by papampi** [comment]
We can also restart network-manager as well when restarting 3main, not tested, no idea if it solve the issue faster or not

**Comment by papampi** [comment]
Checked your email?

**Comment by LuKePicci** [comment]
Try adding in "NetworkManager->WiredConnection>IPv4>Additional DNS Servers" a public dns like 8.8.4.4 or 1.1.1.1 and see if this help.
What sounds really strange to me is that it become functional again after a lot of time ( 3 minutes is not a short time ).

**Comment by papampi** [comment]
Already done that last week, all has dns "192.168.1.1, 1.1.1.1, 8.8.8.8"
Some didnt have router ip in dns list and they were showing this problem more, so added it to all but still some has this problem if network disconnects more than 10-15 minutes.

**Comment by LuKePicci** [comment]
Ok, dnsmasq running on localhost (127.0.0.1) should be automaticaly instructed to forward to dns provided by dhcp. If your dhcp server is sending to all rigs as dns to use 192.168.1.1 and it somehow caches the failure then it may return a cannot resolve error. Try editing that field in network manager but avoiding inserting again 192.168.1.1 as an alternative since it isa already what dhcp sets as default. Conversely, if your dhcp is instructed to send something different, let's say 8.8.8.8, try setting it back to send 192.168.1.1.

**Comment by papampi** [comment]
All rigs has static ip and dns, those without 192.168.1.1 in DNS list was showing this symptom, if I remove 192.168.1.1 all will start showing this problem again.
From 6 rigs without 192.168.1.1, only 2 still has this problem.
Have no local access to them now, will change their dns in next visit.

**Comment by LuKePicci** [comment]
Do you feel that using only 192.168.1.1 as the only dns on all rigs would solve the problem? My rig use dhcp and its dns is also my gateway, and never experienced such a problem,
This PR has a potential pitfall: if the miner is still working fine and the internet check loop fails to connect to google with netcat for some other reasons but ping works, it is restarted without any need.

**Comment by papampi** [comment]
I will check setting dns to only gateway in next visit.
As for potential pitfall, in first check before loop starts if netcat can't resolve google then probably  miner can't resolve pool too and won't start mining.
And in the main loop network check only kicks in if utilization is low and miner shows problems.
Do I miss some thing else?

**Comment by LuKePicci** [comment]
If we assume every miner freeing up gpu resources while network connection is failing then this is fine. Does this assumption holds?

Da: Payam Nab
Inviato: mercoledì 4 luglio 2018 14:46
A: papampi/nvOC_by_fullzero_Community_Release
Cc: Luca Piccirillo; Review requested
Oggetto: Re: [papampi/nvOC_by_fullzero_Community_Release] Solve DNS notresolve after network comes up. (#116)

I will check setting dns to only gateway in next visit.
As for potential pitfall, in first check before loop starts if netcat can't resolve then miner can't resolve too and won't start mining.
And in the main loop network check only kicks in if utilization is low and miner shows problems.
Do I miss some thing else?

**Comment by papampi** [comment]
Miner should free up gpu resources when it get no new jobs, I remember there was a bug in ethminer 14 and it was not freeing up gpus when no work was received because of no network and they fixed it.
All the miners I have seen drop GPU utilization after 2-3 minutes of no network.

File: 5watchdog [hunk 1]
```diff
@@ -158,8 +158,28 @@ fi
...
+    if ping -c1 8.8.8.8 &> /dev/null && ! nc -vzw1 google.com 443
+    then
+      echo " Name resolution problem while ping is ok, waited 10 minutes, restarting system in 10 sec"
+      sleep 10
+      sudo reboot
+    elif ping -c1 8.8.8.8 &> /dev/null && nc -vzw1 google.com 443
+    then
+      echo "Ping and DNS are OK"
+    fi
+  fi
```
Comment (LuKePicci) [inline 5watchdog]: uhm… it is far too complex, simplify this by doing:
```
DNS=0
while ! nc …
do
  if (( DNS >= 20 ))
  then
    echo … waited 10 minutes ...
    reboot
    exit
  fi
  echo … failed test connection ...
  if ping …
  then
    echo … but ping is ok ...
    (( DNS++ ))
  fi
  sleep ...
done
```

**Review by LuKePicci — CHANGES_REQUESTED** [review]

File: 5watchdog (hunk 1 above)
Comment (papampi) [reply 5watchdog]: 👍

**Review by LuKePicci — APPROVED** [review]

**Pull request merged**